services/explorer-indexer/src/modules/pbft/pbft.service.ts

```diff
@@ -121,18 +130,31 @@ export default class PbftService {
       },
     });
     if (pbfts?.length > 0) {
-      await Promise.all(
-        pbfts.map(async (pbft: PbftEntity) => {
+      try {
+        for (const pbft of pbfts) {
           await this.dagService.findAndRemoveDagsForPbftPeriod(pbft.number);
-        })
-      );
-
-      this.logger.debug(
-        `Deleting ${pbfts?.length} PBFTs with number greater or equal than ${pbftNumber}`
-      );
+          if (pbft.transactions && pbft.transactions?.length > 0) {
+            await this.txService.deleteTransactions(
```

Comment:
Are you sure this won't crash anything? In this step the transactions are associated with Dags. That's why they are deleted when Dags are deleted. This might trigger a pg error, either at this step or when trying to delete and cascade Dags. We need to check at least.